### PR TITLE
Harrison/use output parser

### PR DESCRIPTION
**Pull request by hwchase17**
introduce output parser and use it in a for loop

### CODE REVIEW

File: langchain/chains/llm_for_loop.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from langchain.chains.llm import LLMChain
+from langchain.prompts.base import ListOutputParser
+
+
+class LLMForLoopChain(Chain, BaseModel):
+    """Chain that first uses an LLM to generate multiple items then loops over them."""
+
+    llm_chain: LLMChain
+    """LLM chain to use to generate multiple items."""
+    apply_chain: Chain
```
Comment (sjwhitmore) [inline langchain/chains/llm_for_loop.py:16]: can this be a list of chains? and guarantee that the number of generated items is equal to the length of the list?

File: langchain/prompts/base.py [hunk 2]
```diff
@@ -32,11 +32,35 @@ def check_valid_template(
...
 
+class BaseOutputParser(ABC):
+    """Class to parse the output of an LLM call."""
+
+    @abstractmethod
+    def parse(self, text: str) -> Union[str, List[str], Dict[str, str]]:
+        """Parse the output of an LLM call."""
+
+
+class ListOutputParser(ABC):
```
Comment (sjwhitmore) [inline langchain/prompts/base.py:43]: I honestly might prefer parser to be a bit less defined -- e.g. not sure there is a need for a ListOutputParser primitive -- feels like the user could fully define their own OutputParser

Comment (hwchase17) [reply langchain/prompts/base.py:43]: oh they def can! but some chains (like the for loop chain) will expect a list output...

**Comment by dev2049** [comment]
@hwchase17 worth landing for loop chain or can close?

**Pull request closed without merging**